extractor/src/test/java/com/example/extractor/MainsLauncherTest.java

```diff
@@ -143,4 +147,78 @@ private String loadToString(final String filePath) throws IOException {
         return new String(encoded, StandardCharsets.UTF_8);
     }
+
+    @Test
+    void testMainWithExternalConfigFiles(@TempDir final Path tempDir) throws Exception {
+        // Create the 'config' directory inside the tempDir
+        Path configDir = tempDir.resolve("config");
+        Files.createDirectories(configDir);
+
+        // Write the java.header file inside the 'config' directory
+        String headerFileContent = """
+                ^// Copyright \\(C\\) (\\d\\d\\d\\d -)? 2004 MyCompany$
+                ^// All rights reserved$
+                """;
+        Path headerFile = configDir.resolve("java.header");
+        Files.writeString(headerFile, headerFileContent);
+
+        // Write the config.xml file
+        String configXmlContent = """
+                <?xml version="1.0"?>
+                <!DOCTYPE module PUBLIC
+                    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
+                    "https://checkstyle.org/dtds/configuration_1_3.dtd">
+                <module name="Checker">
+                  <module name="RegexpHeader">
+                    <property name="headerFile" value="config/java.header"/>
+                    <property name="multiLines" value="10, 13"/>
+                  </module>
+                </module>
+                """;
+        Path configXmlFile = tempDir.resolve("config.xml");
+        Files.writeString(configXmlFile, configXmlContent);
+
+        // Write the test input file (could be an empty Java file)
+        String testFileContent = """
+                // Sample header content
+                // That should be matched
+
+                public class TestClass {
+                    // class content
+                }
+                """;
+        Path testFile = tempDir.resolve("TestClass.java");
+        Files.writeString(testFile, testFileContent);
+
+        // Define the output files for the extractor
+        Path outputConfigFile = tempDir.resolve("output-config.xml");
+        Path outputProjectsFile = tempDir.resolve("output-projects.yml");
+
+        // Modify the CheckstyleExampleExtractor to accept a config file
+        assertDoesNotThrow(() -> {
+            CheckstyleExampleExtractor.main(new String[]{
+                    CHECKSTYLE_REPO_PATH,
+                    "--config-file",
+                    configXmlFile.toString(),
+                    outputConfigFile.toString(),
+                    outputProjectsFile.toString(),
+            });
+        });
+
+        // Verify that the output config file is created
+        assertTrue(Files.exists(outputConfigFile), "Config output file should be created");
+
+        // Load the generated config content
+        String generatedConfigContent = Files.readString(outputConfigFile);
+
+        // Verify that the generated config includes the external file reference
+        assertThat(generatedConfigContent)
+                .contains("<property name=\"headerFile\" value=\"config/java.header\"/>");
```

Comment:
We need path that will work for diff tool.
I think we can `value=\"${execution.path}/config/java.header\"/>"`
And we will set this variable when we start diff tool, so internal checkstyle execution will not have problems to find this header file.

Reply:
done